docker/automations/bots/bath-lights.js

```diff
@@ -32,6 +38,136 @@ module.exports = (name, {
         let toggledTimer = null
         let unlockedTimer = null
 
+        // Command verification configuration (safe defaults = disabled)
+        const config = {
+            verification: commandConfig.verification || 0,     // 0 = disabled
+            maxRetries: commandConfig.maxRetries || 0,         // 0 = disabled
+            retryDelay: commandConfig.retryDelay || 1000,      // 1 second between retries
+        }
+
+        // Pending commands tracking (only used if verification enabled)
+        const pendingCommands = new Map() // reason -> {expectedState, attempts, timer, etc}
+
+        // Smart publish: legacy mode (direct) or verification mode
+        const smartPublish = (topic, payload, reason = 'unknown') => {
+            if (config.verification === 0 || config.maxRetries === 0) {
+                // Legacy mode - direct publish (existing behavior)
+                mqtt.publish(topic, payload)
+                return
+            }
+
+            // Verification mode - use state-based verification
+            const expectedState = payload.state
+            
+            // Cancel any existing command for the same reason to avoid conflicts
+            if (pendingCommands.has(reason)) {
+                const existingCommand = pendingCommands.get(reason)
+                clearTimeout(existingCommand.verificationTimer)
+                clearTimeout(existingCommand.retryTimer)
+                if (verbose) {
+                    console.log(`[${name}] cancelling existing command for reason: ${reason}`)
+                }
+            }
+
+            const command = {
+                topic,
+                payload,
+                expectedState,
+                reason,
+                attempts: 0,
+                timestamp: Date.now()
+            }
+
+            const executeCommand = () => {
+                command.attempts++
+                
+                if (verbose) {
+                    console.log(`[${name}] sending command (attempt ${command.attempts}): ${JSON.stringify(command.payload)} for reason: ${reason}`)
+                }
+
+                try {
+                    mqtt.publish(topic, command.payload)
+                } catch (error) {
+                    if (verbose) {
+                        console.log(`[${name}] command publish failed for reason ${reason}:`, error.message)
+                    }
+                    // Schedule retry for publish failures
+                    scheduleRetry()
+                    return
+                }
+
+                // Set verification timeout
+                const verificationTimer = setTimeout(() => {
+                    if (pendingCommands.has(reason)) {
+                        if (verbose) {
+                            console.log(`[${name}] command verification timeout for ${reason} (expected: ${expectedState}, actual: ${lightState})`)
+                        }
+                        if (lightState === expectedState) {
```

Comment:
in what scenario this is possible - when lightState change there is a check for pending commands and that should have already cleaned this command